packages/openapi/src/decorators.ts

```diff
@@ -214,3 +218,52 @@ export function resolveInfo(program: Program, entity: Namespace): AdditionalInfo
 function omitUndefined<T extends Record<string, unknown>>(data: T): T {
   return Object.fromEntries(Object.entries(data).filter(([k, v]) => v !== undefined)) as any;
 }
+
+function validateObject(context: DecoratorContext, typespecType: TypeSpecValue) {
+  const propertyModel = context.program.resolveTypeReference(
+    "TypeSpec.OpenAPI.AdditionalInfo",
+  )[0]! as Model;
+
+  if (typeof typespecType === "object" && propertyModel) {
+    const diagnostics = checkNoAdditionalProperties(
+      typespecType,
+      context.getArgumentTarget(0)!,
+      propertyModel,
+    );
+    context.program.reportDiagnostics(diagnostics);
+  }
+}
+
+function checkNoAdditionalProperties(
+  typespecType: Type,
+  target: DiagnosticTarget,
+  source: Model,
+): Diagnostic[] {
+  const diagnostics: Diagnostic[] = [];
+
+  if (typespecType.kind === "Model") {
```

Comment:
can this be anything else than a model, if not better to use `compileAssert(typespecType.kind === "Model"`